### DISCUSSION TITLE
Reflection is not able to wrap/unwrap an object from his interface

### DISCUSSION THREAD

**Discussion by Dracks** [vlang/v#18683] [Ideas]
### Describe the bug

Code: https://vosca.dev/p/d563469814

```v
struct Something {
	count int = 0
}

struct StrWithSomething {
	mut:
	something &Something = unsafe{nil}
}

interface Object {}

fn main(){
	s := &Object(Something{})
	mut str_with := StrWithSomething{}
	$for field in StrWithSomething.fields {
		str_with.$(field.name) = s
	}
	println(str_with)
}


```

### Expected Behavior

In this case, s should be unpacked from the Object interface, to assign correctly the original struct to the property.

### Current Behavior

is assigning the C Object (representing the V interface) directly to the field, not detecting the change of type on the progress. 

The output field, you get allways a count != 0 (as it's pointing to the wrong place in memory)

Output:

```
StrWithSomething{
    something: &Something{
        count: 340525024
    }
}
```

### Reproduction Steps

Simply run the program, and you will see it. 

### Possible Solution

when doing the reflection, V should detect the type of the field at which are we accessing, and process it as normal code 

### Additional Information/Context

I'm using the following version:
V 0.3.4 e2f18fc

### V version

V 0.3.3 fc4c431.e738d67

### Environment details (OS name and version, etc.)

```
V full version: V 0.3.4 fc4c431.099d4fc
OS: linux, Ubuntu 22.04.2 LTS
Processor: 2 cpus, 64bit, little endian, Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz

getwd: /home/pmakhnev/playground
vexe: /home/pmakhnev/v/v
vexe mtime: 2023-05-24 06:00:08

vroot: OK, value: /home/pmakhnev/v
VMODULES: OK, value: /root/.vmodules
VTMP: OK, value: /tmp/v_0

Git version: git version 2.34.1
Git vroot status: weekly.2023.21-5-g099d4fc0 (2 commit(s) behind V master)
.git/config present: true

CC version: cc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3
```

**Comment by felipensp** [comment]
I don't get it. If you do `str_with.something = s` it produces: 
```
cannot assign to `str_with.something`: expected `&Something`, not `&Object`
```
Why using comptime reflection should it works?

**Comment by Dracks** [comment]
ok, I was thinking I tested that case and was working. I see that it's not working, It will be good at least to raise an error if v doesn't wish to be able to handle. I think will be really usefull to be able to do the cast/wrap, correctly.
